stable/gocd/templates/gocd-server-deployment.yaml

```diff
@@ -57,6 +57,7 @@ spec:
       {{- if .Values.server.security.ssh.enabled }}
         - name: ssh-secrets
           secret:
+            defaultMode: 256
```

Comment:
We can do that, but it will not work with any other permissions. We will face [this](https://user-images.githubusercontent.com/6998650/80254499-a2b68400-86be-11ea-9205-358c608dcc72.png) error if the permission is other than `256`.

Reply:
The ssh directory may be mounted for reasons other than using it with a private git material. This overrides the original file permissions on all the files in the ssh directory before mounting. So providing users an option to change the default mode seems better to me.

Reply:
Cool, that makes sense. I will make the changes now.